### PR TITLE
BUG: fixed index power operation

### LINKED ISSUE #14973: Using the python power operator on numerical Index objects yields unexpected results
#### Code Sample, a copy-pastable example if possible

```python
index = pd.Float64Index(range(1, 11))

expected_pow_values = 2.0**index.values
actual_pow_values = 2.0**index
similar_pow_values = 2.0**np.arange(len(index))

expected_mul_values = 2.0*index.values
actual_mul_values = 2.0*index

other_way_around = index**2.0
same_as = index.values**2.0

```
#### Problem description

Using the pow '**' operator on an index object results in a behaviour different than when using the '*' operator. The pow result implies that the values being used are not index.values, which one would expect given the behaviour of other operators (*, -, +, / ...).
This problem only arises when the index object is the second argument.

#### Expected Output
expected output would be the result of 2.0**index.values

#### Output of ``pd.show_versions()``

<details>

commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.19.1
nose: None
pip: 9.0.1
setuptools: 32.2.0
Cython: None
numpy: 1.12.0rc1
scipy: 0.18.1
statsmodels: 0.8.0rc1
xarray: None
IPython: 5.1.0
sphinx: 1.4.8
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.3
openpyxl: 2.4.1
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: None
lxml: 3.7.0
bs4: None
html5lib: 0.9999999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None

</details>

### PR DESCRIPTION
**Pull request by kamal94**
* The power operation on a range of indexes was fixed. See issue #14973

 - [ ] closes #14973
 - [ ] tests added / passed
 - [ ] passes ``git diff upstream/master | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/test_numeric.py [hunk 1]
```diff
@@ -105,6 +105,21 @@ def test_numeric_compat(self):
         for r, e in zip(result, expected):
             tm.assert_index_equal(r, e)
 
+        #test power calculations both ways
```
Comment (jreback) [inline pandas/tests/indexes/test_numeric.py]: model this after the existing code, IOW, use ``idx`` which is already define.

add a comment with this issue number.

File: pandas/tests/indexes/test_numeric.py [hunk 2]
```diff
@@ -105,6 +105,21 @@ def test_numeric_compat(self):
...
+        num_range = np.arange(1,11)
+        pow_range = 2.0**num_range
+        idxs = pd.Float64Index(range(1, 11))
+
+        expected_pow_values = pd.Float64Index(pow_range)
```
Comment (jreback) [inline pandas/tests/indexes/test_numeric.py]: use ``expected`` and ``result``

File: doc/source/whatsnew/v0.20.0.txt [hunk 3]
```diff
@@ -282,6 +282,7 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
 
+- Bug in ``Indexes`` Power numerical operations did not operate correctly (:issue:`14973`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.20.0.txt]: Bug in ``Index`` power operations with with reversed operands

File: pandas/tests/indexes/test_numeric.py (hunk 1 above)
Comment (kamal94) [inline pandas/tests/indexes/test_numeric.py]: Thank you for your guidance. Changes implemented.

**Comment by kamal94** [comment]
@jreback  anything more I can do to have it accepted?

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**